src/main/java/uk/gov/companieshouse/api/accounts/service/impl/AccountServiceImpl.java

```diff
@@ -0,0 +1,36 @@
+package uk.gov.companieshouse.api.accounts.service.impl;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DataAccessException;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import uk.gov.companieshouse.api.accounts.model.entity.AccountEntity;
+import uk.gov.companieshouse.api.accounts.model.rest.Account;
+import uk.gov.companieshouse.api.accounts.repository.AccountRepository;
+import uk.gov.companieshouse.api.accounts.service.AccountService;
+
+@Service
+public class AccountServiceImpl implements AccountService {
+
+    @Autowired
+    private AccountRepository accountRepository;
+
+    /**
+     * {@inheritDoc}
+     */
+    public ResponseEntity createAccount(Account account) {
+        try {
+            AccountEntity accountEntity = new AccountEntity().build(account);
+
+            AccountEntity createdAccount = accountRepository.insert(accountEntity);
+            if (createdAccount != null) {
+                return ResponseEntity.status(HttpStatus.CREATED).body(createdAccount.getData());
+            } else {
+                return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
+            }
+        } catch (DataAccessException dataAccessException) {
+            return new ResponseEntity(HttpStatus.INTERNAL_SERVER_ERROR);
```

Comment:
The service should not be setting HTTP response codes. It should only be handling the logic. The controller has knowledge of HTTP requests and responses.

Reply:
What would you recommend as a change?

Reply:
Return a boolean for whether the creation was successful from the service and set the appropriate response header in the controller. The exception/failure reason can be logged from the service.

Reply:
How would we then return the data of the account?

Reply:
Sorry, didn't spot the entire account being returned. You could return the account from the service and check if it is null in the controller for setting the HTTP codes.

The returned account from the service won't contain the etag, kind and links because you are setting them in the build method in the DB entity?

Reply:
So you want it to return the rest object instead?

Reply:
Yes